src/libexpr/parser.y

```diff
@@ -206,9 +216,21 @@ expr_function
 
 expr_if
   : IF expr THEN expr ELSE expr { $$ = new ExprIf(CUR_POS, $2, $4, $6); }
+  | expr_apply
+  | expr_apply_flipped
   | expr_op
   ;
 
+expr_apply
```

Comment:
what's the difference between `expr_apply` and `expr_app`? It's a bit confusing to me.

Reply:
Perhaps we could do

```suggestion
expr_pipe_apply
```

and `expr_pipe_apply_flipped`, `expr_pipe_into` or `expr_pipe_forward` for the other production.

Reply:
I've renamed the tokens and non-terminals. Please let's not get hung up on making the names perfect; these are internal names that can be improved at any time, whereas releasing this feature is something that other people are waiting for.